homework/week1/2025320048-yujuyeon/[Homework][Week1] 1027번.c

```diff
@@ -0,0 +1,53 @@
+#include <stdio.h>
+typedef struct {
+    int h;
+    int num;
+} bldg;
+
+int left(bldg b[], int idx) {
+    double incli = 0;
+    double min_incli = 1000000001;
+    int renew = 0;
+    for (int i = 1; idx - i >= 0; i++) {
+        incli = ((double)(b[idx].h - b[idx - i].h)) / i; //연산 과정에서의 자료형이 중요함... *명시적 형 변환
+        if (incli < min_incli) {
+            min_incli = incli;
+            renew++;
+        }
+    }
+    return renew;
+}
+int right(bldg b[], int idx, int max_idx) {
+    double incli = 0;
+    double max_incli = -1000000001;
+    int renew = 0;
+    for (int i = 1; idx + i <= max_idx; i++) {
+        incli = ((double)(b[idx + i].h - b[idx].h)) / i;
+        if (incli > max_incli) {
+            max_incli = incli;
+            renew++;
+        }
+    }
+    return renew;
+}
```

Comment:
left랑 하는 역할이 매우 유사해보입니다. 유사하지만, 다른 함수로 만들면 여러 포인트를 관리하고, 버그가 생길 확률이 높아집니다.
혹시 main에서 아래와 같이 사용해보시는건 어떠신가요?
```suggestion
    bldg inverse_building[50];
    for(int i = 0; i < n; i++) {
        inverse_building[i].h = b[n - 1 - i].h;
    }
    for(int i = 0; i < n; i++) {
        b[i].num = left(b, i) + left(inverse_building, n - 1 - i);
        // 다른 코드들
    }
```